src/sparseml/exporters/transforms/kv_cache/cache_keys_and_values.py

```diff
@@ -200,40 +202,46 @@ def create_cache(
         this is -2, which corresponds to the sequence length axis.
     :param use_uint8_if_quantized: True if quantized MatMuls should have uint8
         inputs, if False, uses int8
+    :param batch_size: batch size of the kv cache. By default, this is 1.
     :return: tuple of concat node to add, cache input to add, and cache output to add,
         updates existing nodes in-place
     """
-    graph = ONNXGraph(model)
-
-    if node.op_type == "QLinearMatMul" and cache_input_idx == 1:
-        cache_input_idx = 3  # QLinearMatMul B matrix is at idx 3, not 1
-
-    cache_parent = graph.get_node_single_parent(node, index=cache_input_idx)
-    if isinstance(cache_parent, NodeProto) and cache_parent.op_type == "Transpose":
-        # move cache to before a transpose if applicable
-        # this is due to pytorch operations potentially extracting shape values
-        # from the key tensor before the transpose is applied
-        pre_cache_input_id = cache_parent.input[0]
-        # update concat axis
-        node = cache_parent
-    else:
-        pre_cache_input_id = node.input[cache_input_idx]
-
-    concat_node = onnx.helper.make_node(
-        op_type="Concat",
-        inputs=[cache_input_name, pre_cache_input_id],
-        outputs=[cache_output_name],
-        axis=concat_axis,
-        name=f"concat.{cache_input_name}",
-    )
-
-    cache_input_dims = [num_attention_heads, "past_sequence_len", hidden_size_kv_cache]
+    cache_input_dims = [
+        batch_size,
+        num_attention_heads,
+        "past_sequence_len",
+        hidden_size_kv_cache,
+    ]
     cache_output_dims = [
+        batch_size,
         num_attention_heads,
         "past_sequence_len + 1",
         hidden_size_kv_cache,
     ]
 
+    cache_input_name_reshaped = f"{cache_input_name}_reshaped"
```

Comment:
getting a bit long now - let's move this cache reshape node(s) injection to a helper function

Reply:
Yes, intended do it on the CodeGen KV Cache PR, but it fits better here.